crates/hir-def/src/nameres/path_resolution.rs

```diff
@@ -159,13 +160,15 @@ impl DefMap {
                 (None, new) => new,
             };
 
-            match &current_map.block {
-                Some(block) => {
+            match current_map.block {
+                Some(block) if original_module == Self::ROOT => {
```

Comment:
is the guard necessary? Aren't block modules always the root of their block def map?

Reply:
Yes, but the thing is, block `DefMap`s may also have `mod`s, which *are* module boundary and we don't want to continue this loop. Rust2 needs to ban `mod`s inside blocks because why would anyone want them? :roll_eyes:

Reply:
right :facepalm: